### PR TITLE
Support executing a single test function

### LINKED ISSUE #24628: Support executing a single test function
**Description:**

Running single test cases in the current test framework is not possible. It may be brought about by including groups or disabling tests but this is a cumbersome task.

### PR DESCRIPTION
**Pull request by Dilhasha**
Fixes #24628 

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [x] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/TestCommand.java [hunk 1]
```diff
@@ -149,6 +149,9 @@ public TestCommand(Path userDir, PrintStream outStream, PrintStream errStream, b
     @CommandLine.Option(names = "--home-cache", description = "Custom home cache")
     private String homeCache;
 
+    @CommandLine.Option(names = "--functions", split = ",", description = "Test functions to be executed")
```
Comment (hevayo) [inline cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/TestCommand.java:152]: Can we name the options as --tests

File: tests/testerina-integration-test/src/test/java/org/ballerinalang/testerina/test/SelectedFunctionTest.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        String msg = "2 passing";
+        LogLeecher clientLeecher = new LogLeecher(msg);
+        balClient.runMain("test", new String[]{"--functions", "testFunc,testFunc2",
+                "beforeEachAfterEach"}, null, new String[]{}, new LogLeecher[]{clientLeecher}, projectPath);
+        clientLeecher.waitForText(20000);
+    }
+
+    @Test
+    public void testNonExistingFunctionExecution() throws BallerinaTestException {
+        String msg = "No tests found";
```
Comment (hevayo) [inline tests/testerina-integration-test/src/test/java/org/ballerinalang/testerina/test/SelectedFunctionTest.java:69]: Can we give a more specific response

File: tests/testerina-integration-test/src/test/java/org/ballerinalang/testerina/test/SelectedFunctionTest.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        LogLeecher clientLeecher = new LogLeecher(msg);
+        balClient.runMain("test", new String[]{"--functions", "nonExistingFunc", "--all"},
+                null, new String[]{}, new LogLeecher[]{clientLeecher}, projectPath);
+        clientLeecher.waitForText(20000);
```
Comment (hevayo) [inline tests/testerina-integration-test/src/test/java/org/ballerinalang/testerina/test/SelectedFunctionTest.java:73]: What is the error code of this.

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/TestCommand.java (hunk 1 above)
Comment (azinneera) [inline cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/TestCommand.java:152]: Wouldnt it be better to name the option as `--tests`

File: tests/testerina-integration-test/src/test/java/org/ballerinalang/testerina/test/SelectedFunctionTest.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+import org.ballerinalang.test.context.BMainInstance;
+import org.ballerinalang.test.context.BallerinaTestException;
+import org.ballerinalang.test.context.LogLeecher;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+/**
+ * Test class containing tests related to selective function tests.
+ */
+public class SelectedFunctionTest extends BaseTestCase {
```
Comment (azinneera) [inline tests/testerina-integration-test/src/test/java/org/ballerinalang/testerina/test/SelectedFunctionTest.java:29]: Shall we add test cases to check the behavior to test the following:
1. when trying to run a disabled test case
2 when trying to run a test case which depends on a disabled tests case

File: tests/testerina-integration-test/src/test/java/org/ballerinalang/testerina/test/SelectedFunctionTest.java (hunk 3 above)
Comment (Dilhasha) [reply tests/testerina-integration-test/src/test/java/org/ballerinalang/testerina/test/SelectedFunctionTest.java:75]: If the provided text is not found within the timeout period, an error message as "Timeout expired waiting for matching log" will be given. Does that answer your question?

Comment (azinneera) [reply tests/testerina-integration-test/src/test/java/org/ballerinalang/testerina/test/SelectedFunctionTest.java:75]: AFAIU, @hevayo is asking about the exit code of the command. According to the current implementation, the exit code is `0`, when no tests found to execute. Ideally this should be `1`.

Comment (hevayo) [reply tests/testerina-integration-test/src/test/java/org/ballerinalang/testerina/test/SelectedFunctionTest.java:75]: Yeh sorry I was referring to exit code

**Pull request merged**